Server/src/models/DataUploadModel.ts

```diff
@@ -15,190 +15,185 @@ import { Material } from './entities/Material';
+    async insertReferenceType(preferenceType: string) {
 
-    let book = new Publicationtype();
-    book.id;
-    book.name = preferenceType;
-    await connection.manager.save(book);
-    return book.id;
-}
+        const connection = getConnection();
+
+        let book = new Publicationtype();
+        book.id;
+        book.name = preferenceType;
+        await connection.manager.save(book);
+        return book.id;
+    }
 
-const insertPublisher = async (publisher) => {
+    async insertPublisher(publisher: string) {
 
-    const connection = getConnection();
+        const connection = getConnection();
 
-    let publisherName = new Publisher();
-    publisherName.id;
-    publisherName.name = publisher;
-    await connection.manager.save(publisherName);
-    return publisherName.id;
-}
+        let publisherName = new Publisher();
+        publisherName.id;
+        publisherName.name = publisher;
+        await connection.manager.save(publisherName);
+        return publisherName.id;
+    }
 
-const insertAuthors = async (storeAuthors) => {
+    async insertAuthors(storeAuthors: any[]) {
 
-    const connection = getConnection();
+        const connection = getConnection();
 
-    for (var i = 0; i < storeAuthors.length; i++) {
-        console.log(storeAuthors[i].firstname);
-        let author = new Authors();
-        author.id;
-        author.firstName = storeAuthors[i].firstname;
-        author.lastName = storeAuthors[i].lastname;
-        author.middleName = storeAuthors[i].middlename;
-        await connection.manager.save(author);
+        for (let i = 0; i < storeAuthors.length; i++) {
+            console.log(storeAuthors[i].firstname);
+            let author = new Authors();
+            author.id;
+            author.firstName = storeAuthors[i].firstname;
+            author.lastName = storeAuthors[i].lastname;
+            author.middleName = storeAuthors[i].middlename;
+            await connection.manager.save(author);
+        }
     }
-}
 
-const insertPublication = async (referenceTitle, referencePages, preferenceTypeID, publisherNameId, referenceYear, referenceVolume, referenceAuthors) => {
-
-    const connection = getConnection();
-
-    let publication = new Publications();
-    publication.id;
-    publication.name = referenceTitle;
-    publication.doi;
-    publication.pages = referencePages;
-    publication.publicationtypeId = preferenceTypeID;
-    publication.publisherId = publisherNameId;
-    publication.year = referenceYear;
-    publication.volume = referenceVolume;
-    publication.datePublished;
-    publication.dateAccessed;
-    publication.authors = referenceAuthors;
-    await connection.manager.save(publication);
-    console.log("publication saved.");
-    return publication.id;
-}
+    async insertPublication(referenceTitle: string, referencePages: number, preferenceTypeID: number, publisherNameId: number, referenceYear: number, referenceVolume: number, referenceAuthors: any[]) {
```

Comment:
_ideally_ we should avoid using 'any' type as much as possible, **especitally** when it is objects that we produce, referenceAuthors should have be of type IAuthorModel which is in turn an interface that encapsulates all the field of the author (or the ones that are needed):
export interface IAurthorModel {
...
fields: number
...
}

Reply:
This is a good suggestion will update